app/models/discussion.rb

```diff
@@ -13,8 +13,6 @@ class Discussion < ActiveRecord::Base
   private
 
   def update_subscriptions
-    User.all.each do |user|
-      Subscription.for(discussion_id: id, user_id: user.id).new_comment
-    end
+    Subscriber.for_discussion(self).subscribe(User.all)
```

Comment:
Right, I forgot about this. This was in reality doing two things, which were hidden by the `Subscription.for` method. It was either subscribing a user to a discussion and then moving it to the `new_comment` state, or getting an existing one and moving it to the same state.

I think we can make the `subscribe` method return a list of subscriptions either newly created or existent, and then:

```ruby
subscriptions.map(&:new_comment)
```

What do you think?